app/models/merchant.rb

```diff
@@ -10,4 +11,13 @@ def revenue(filter = nil)
     .joins(:invoice_items, :transactions)
     .sum("quantity * unit_price")
   end
+
+  def self.favorite_customer(filter = nil)
+    results =
```

Comment:
This doesn't look bad to me, you'll have to let me know what you mean about refactoring.